sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs

```diff
@@ -54,7 +54,7 @@ public string CredentialWords
                     .Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries)
                     .Select(x => x.Trim())
                     .ToList();
-                this.passwordValuePattern = new Regex(string.Format(@"\b(?<password>{0})\b[:=]\S",
+                this.passwordValuePattern = new Regex(string.Format(@"\b(?<password>{0})\b[:=](?!\?|:|\{{[0-9]*\}})\S",
```

Comment:
I don't understand this part:
```
\{{[0-9]*\}}
```
To break it into pieces, I see:
```
\{         Escaped character { - OK

{          Start of Quantifier? The quantifier format should be like {1}, {1,2} or {1,}. Maybe it's some escape sequence I don't know, can you provide documentation for that?
  [0-9]*   Zero or more numbers. I prefer \d for readability. I think "zero" is mistake, it should be at least one, right? So \d+ instead.
  \}       Escaped character } - why is it inside the { } Quantifier section? First character was outside.
}          End of Quantifier? Same question as Start of Quantifier.
```

Reply:
In this case (`password={0}`), `{0}` can be any number, or, as you already mentioned, an explicit format.